src/aof.c

```diff
@@ -77,9 +78,8 @@ off_t getBaseAndIncrAppendOnlyFilesSize(aofManifest *am);
 #define INCR_FILE_SUFFIX           ".incr"   
 #define RDB_FORMAT_SUFFIX          ".rdb"
 #define AOF_FORMAT_SUFFIX          ".aof"
-#define MANIFEST_NAME_SUFFIX       ".manifest"   
+#define MANIFEST_NAME_SUFFIX       "_manifest"   
```

Comment:
Because I want to make them more unified:
```
appendonly.aof_1.base.aof 
appendonly.aof_1.incr.aof 
appendonly.aof_manifest
```

Reply:
ok, in my eyes it's a table:
```
base-prefix      idx   role   format
appendonly.aof   1     base   aof 
appendonly.aof   1     incr   aof 
appendonly.aof                manifest
```
maybe if we change the `_` to `.`, or change the first `.` to `_` it'll make more sense?
i.e. either:
`appendonly.aof.1.base.aof`
or
`appendonly.aof_1_base.aof`

Reply:
I don't like to have too many `.` ,  it can easily lead to misunderstandings because it contains the `.aof` `.1` `.base` `.aof` suffix

Reply:
I offer a suggestion: 
aof dir:
```
 appendonly.aof_1.base appendonly.aof_1.incr  appendonly.aof_2.incr  appendonly.aof_manifest
```

the content of manifest:
```
file appendonly.aof_1.base seq 1 type b format rdb
file appendonly.aof_1.incr seq 1 type i format aof
file appendonly.aof_2.incr seq 2 type i format aof
```

Reply:
i don't see the problem with multiple `.` in the file name.. it's not uncommon, and it's a well established pattern to only look past the last `.` for the file format, so in that respect, in my eyes the manifest is the format and should use `.`
also, there are already two `.` in your other file patterns, so i'm not sure what's bothering you about it.
if the other `_` bother us, we can change it to `.` as well.

as for the format suffix, i don't think it should be part of the manifest, i think it should be in the file name, and what's bothering us is that we have two mentions of the "format" in the file name (`.aof` can appear twice)

Reply:
I think it is necessary for us to vote for this rule. Let me summarize that we now have blow options (@oranagra Maybe you have other plans to add):
1.
aof dir:
```
 appendonly.aof.1.base.aof 
 appendonly.aof.1.incr.aof 
 appendonly.aof.1.incr.aof 
 appendonly.aof.manifest
```

2.
aof dir:
```
 appendonly.aof_1.base.aof 
 appendonly.aof_1.incr.aof 
 appendonly.aof_1.incr.aof 
 appendonly.aof_manifest
```

3.
aof dir:
```
 appendonly.aof_1.base 
 appendonly.aof_1.incr
 appendonly.aof_1.incr
 appendonly.aof_manifest
```

4.
aof dir:
```
 appendonly.aof_1.base 
 appendonly.aof_1.incr  
 appendonly.aof_2.incr  
 appendonly.aof_manifest
```
and the content of manifest:
```
file appendonly.aof_1.base seq 1 type b format rdb
file appendonly.aof_1.incr seq 1 type i format aof
file appendonly.aof_2.incr seq 2 type i format aof
```

@oranagra @yossigo @soloestoy @madolson Please vote which one of the above are more satisfactory to you, or if you have a better suggestion, thank you.

Reply:
i had a discussion about this (the annoying presence of ".aof" in file names derived from `appendfilename`) with @yoav-steinberg to try to see if we can come up with something.

one idea that came up is this:
1. when creating the manifest on upgrade, use `server.aof_filename`, but without any ".aof" suffix if present
2. when creating the base file entry for the upgrade in the manifest, do the same (remove the ".aof" suffix)
3. after when moving the existing AOF file into the folder, do the same.
4. after the upgrade is done (rename is completed), and on normal startup if there was no upgrade, we actually trim the ".aof" suffix from `server.aof_filename` (modify the variable)

so now, if someone does `CONFIG GET appendfilename` or `CONFIG REWRITE`, the value of the config no longer has the `.aof` suffix.

so if i had a concern that that someone will write a bash script doing
```
cp `config get appenddirname`/`config get appendfilename` <some destination>
```
this would still work.

notes:
* i don't particularly like this idea (yet)
* I must note that we have other configs for which the CONFIG GET value is mutated. e.g. `CONFIG SET maxmemory 1g` will translate to a different value in CONFIG GET.
* it would still look ugly for anyone looking at the redis.conf file that's shipped with redis (mentioning `appendonly.aof`, in case there are no config rewrites), but what we can do is delete that line from the default config, and only rely on the default value hard-coded in redis.
* we can update the comment in redis.conf to just mention `# appendfilename appendonly` (without the suffix)
* this means that the only way a user of a new deployment to observe that the hard coded default value is with `.aof` is by looking at the code, since at startup we change it right after handling the upgrade procedure, that default is not observable in any way.
* unlike the `maxmemory` config, this one is immutable, so you can't observe that you do CONFIG SET, and get a different value in CONFIG GET.

Reply:
I don’t think it’s a idea to modify the user’s configuration without authorization. `appendfilename` and `maxmemory` are different. If we trim `appendfilename`, we won’t be able to distinguish between these two configurations.
`appendfilename appendonly `and `appendfilename appendonly.aof`.

Reply:
i agree..
i don't see any other good way to get rid of ".aof" we have in the middle of the file names we create.

I vote for 1

Reply:
I'm also in favor of 1.

I don't see the `.aof` as such a big problem as it only affects users who transition from an older version and don't bother updating the configuration file. It's important not to do something terribly wrong in this case, but having less-than-ideal file naming seems reasonable to me.

Reply:
I voted for 5, just from my personal aesthetic point of view.

Reply:
First of all, I still don't like the method that using `appendfilename` as prefix, it makes the new files' name ugly and hard to understand. I prefer using file lock (i.e. a unified `LOCK` file in folder), if user start some redis instances in the same folder, just log and exit.

But if you insist on handling some strange Schrodinger's scenarios, I vote for 2.

Because use `.` to combine may lead to a problem, if someone like me don't like the prefix, he or she may set the prefix `appendfilename` to empty, file start with `.` is invisible.

Reply:
i don't see why anyone would set the prefix to empty.
you'd better set it to just "appendonly", or "my_redis", or "redis-1".

anyway, i don't mind to use `_` in between the different parts of the template, but i think the last part should be the file format (obviously using `.`).

so maybe this could be a plan:
```
 appendonly.aof_1_base.aof 
 appendonly.aof_1_incr.aof 
 appendonly.aof_2_incr.aof 
 appendonly.aof_manifest.csv
```

note that the manifest is now a CSV instead of being space separated.
and maybe that brings another concern: whatever we choose ad a separator, we must make sure the user doesn't include in his file names `appendfliename` since it'll mess up our parsing (i.e. if he uses space or comma)

Reply:
Coming in late, I would also prefer option 1, but feel indifferent to option 2.